src/hooks/types/Publisher.ts

```diff
@@ -0,0 +1,9 @@
+export type PublisherState = "ready" | "connecting" | "streaming";
+
+export interface Publisher {
+    startStreaming: (mediaStream: MediaStream) => Promise<void>;
+    stopStreaming: () => void;
+    updateAudioSource: (mediaStream: MediaStream) => Promise<void>;
```

Comment:
maybe just updateAudioTrack and updateVideoTrack

Reply:
Renamed :)